DSharpPlus.Core/Rest/RatelimitBucket.cs

```diff
@@ -0,0 +1,142 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Net.Http.Headers;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace DSharpPlus.Core.Rest
+{
+
+    // represents a single ratelimit bucket, against a single endpoint.
+    internal record RatelimitBucket
+    {
+        private const string UnlimitedRatelimitIdentifier = "unlimited";
```

Comment:
"Unlimited" buckets simply pull from the global bucket unless exempt.

Reply:
unlimited is really only used as a fallback here, not for actual request bucketing